lib/scrabble/scrabble.rb

```diff
@@ -0,0 +1,154 @@
+					return value_letters_array[0]
+				end
+			end
+		end
+
+		def self.score(word)
+			word = word.to_s.downcase
+
+			word_score = 0
+
+			word_array = word.chars.to_a
+
+			if word_array.length > 7
+				return "Please enter a word that has 7 or fewer characters."
+
+			elsif word_array.include?(" ")
+				return "Please only enter one word."
+
+			elsif word_array.length == 0
+				return "Please enter at least one letter."
+
+			else
+				word_array.each do |letter|
+					if ("a".."z").include?(letter)
+						letter_score = Scrabble.value_of(letter)
+
+						word_score += letter_score
+					else
+						return "This game only includes letters a-z."
+					end
+				end
+					return word_score
+			end
+		end
+
+
+		def self.array_of_words_by_letter(array_of_words)
+			array_of_words_by_letter = []
+			
+			array_of_words.each do |word|
+				word = word.to_s.downcase
+
+				word_array = word.chars.to_a
+				array_of_words_by_letter.push(word_array)
+			end
+
+			return array_of_words_by_letter
+		end
+
+		def self.array_of_word_scores(array_of_words)
+			array_of_word_scores = []
+
+			array_of_words.each do |word|
+				word_score = Scrabble.score(word)
+				array_of_word_scores.push(word_score)
+			end
+
+			return array_of_word_scores
+		end
+
+
+		def self.highest_score_from(array_of_words)
+
+			array_of_words_by_letter = Scrabble.array_of_words_by_letter(array_of_words)
+			array_of_word_scores = Scrabble.array_of_word_scores(array_of_words)
+
+			# if one of the words has 7 letters, then it automatically wins
+			array_of_words_by_letter.each do |letter_array|
+				if letter_array.length == 7
+					winning_word = letter_array.join
+					return winning_word
+				end
+			end
+
+			#______________________________________________________
+
+			# ANOTHER WAY OF FINDING THE MAX VALUE THAT I WAS PRETTY PROUD
+			# OF FIGURING OUT, BUT WAS MORE COMPLICATED THAN NECESSARY
+			# AND LED TO OTHER ISSUES, SO I PIVOTED INTO SOMETHING ELSE
+
+			# winning_score = 0
+			# winning_index = 0
+			# winning_scores_array = []
+
+			# # Finds the highest score(and corresponding index) in the word scores array
+			# array_of_word_scores.each_index do |index|
+
+			# 	if array_of_word_scores[index.to_i] > winning_score
+			# 		winning_index = index.to_i
+			# 		winning_score = array_of_word_scores[index.to_i]
+			# 	end
+			# end
+
+			# # checks to see if winning score is a duplicate
+			# array_of_word_scores.each_index do |index|
+			# 	if array_of_word_scores[index.to_i] == winning_score
+			# 		winning_scores_array.push(index)
+			# 	end
+			# end
+
+			# if winning_scores_array.length > 1 # then there are duplicate winners
+			# 	winning_scores_array.each do |index|
+			# 		winning_word_length = array_of_words_by_letter[index.to_i].length
+			# 	end
+			
+			# else
+			# 	winning_word = array_of_words_by_letter[winning_index.to_i].join
+
+			# 	return winning_word
+			# end
+
+			#______________________________________________________
```

Comment:
I like this approach! It's longer, but easy to follow and understand the intent.